merge URLs into an array

hal-shu-sato opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

- merge `downloadURL` and `downloadMirrorURL` into `downloadURLs` as an array.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

- #144 

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] ~~Updated the modification date in `mod.xml`.~~

mitosagi commented on v3/types/packages.d.ts:
```diff
@@ -49,13 +49,9 @@ export interface Packages {
-     * The URL of the mirror of the package download page
-     */
-    downloadMirrorURL?: string;
+    downloadURLs?: string[];
```
After changing the schema, use `yarn fix` again to generate it automatically.

mitosagi commented on v3/schema/packages.json:
```diff
@@ -63,15 +63,13 @@
+            "items": {
+              "type": "string",
+              "pattern": "^https?:\\/\\/"
+            },
```
It is better to specify `"minItems": 1` .

mitosagi commented:
Did you change `v3/data/packages.json` manually? I implemented the conversion to https://github.com/hal-shu-sato/apm/pull/469 for review, but it was difficult to compare because of the different key positions in the Json.

There are some disadvantages, such as the `id` key being moved from the beginning of the object to the middle, but how about introducing [prettier-plugin-sort-json](https://github.com/Gudahtt/prettier-plugin-sort-json) to unify the order?

hal-shu-sato commented:
I fixed them.

> Did you change `v3/data/packages.json` manually? I implemented the conversion to [hal-shu-sato/apm#469](https://github.com/hal-shu-sato/apm/pull/469) for review, but it was difficult to compare because of the different key positions in the Json.

I used these code:

```js
import fs from 'fs-extra';

/**
 * adds downloadURLs and remove downloadURL and downloadMirrorURL
 */
async function run() {
  const path = 'C:\\Development\\apm-data\\v3\\data\\packages.json';
  const object = await fs.readJson(path);

  object.packages.forEach((element) => {
    const newURLs = [];
    newURLs.push(element.downloadURL);
    if ('downloadMirrorURL' in element) {
      newURLs.push(element.downloadMirrorURL);
      delete element.downloadMirrorURL;
    }
    element.downloadURL = newURLs;
    console.log(element.downloadURL);
  });

  const result = JSON.stringify(object).replaceAll(
    'downloadURL',
    'downloadURLs'
  );

  fs.writeFile(path, result);
}

run();
```

> There are some disadvantages, such as the `id` key being moved from the beginning of the object to the middle, but how about introducing [prettier-plugin-sort-json](https://github.com/Gudahtt/prettier-plugin-sort-json) to unify the order?

It is a good idea!
Let's do that.

mitosagi reviewed (approved):
Looks good.

I just noticed that you maintained the sort order to reduce diffs.
